Scripts/Interfaces/Entity/IMainMenu.cs

```diff
@@ -35,5 +35,31 @@ public interface IMainMenu : IMenu, IUsesMenuActions, ISelectTool
 		/// </summary>
 		Transform targetRayOrigin { set; }
 	}
+
+	public static class IMainMenuMethods
+	{
+		public static Action<Transform, Type, string> previewInPinnedToolButton { get; set; }
+
+		/// <summary>
+		/// Highlights a pinned tool button when a menu button is highlighted
+		/// <param name="rayOrigin">Transform: Ray origin to check</param>
+		/// <param name="toolType">Type: MenuButton's tool type to preview</param>
+		/// <param name="toolDescription">String: The tool description to display as a Tooltip</param>
+		/// </summary>
+		public static void PreviewInPinnedToolButton (this IMainMenu obj, Transform rayOrigin, Type toolType, string toolDescription)
```

Comment:
Interesting call... Yeah, this could be helpful as a separate interface.  Say if a workspace, or scene element would want to preview something in a PinnedToolButton.